### PR TITLE
[#34] Add support for email notification to tree owner

### LINKED ISSUE #34: Notify tree owner once "pending" status is removed
Send automatic email to tree owner to indicate that their tree subscription was reviewed and completed.
From https://github.com/tiagovaz/saskatoon/issues/192

### PR DESCRIPTION
**Pull request by victorphoenix3**
Adds support for email notification to tree owner when pending status is removed by administrator.

Fixes #34 
Depends on #213 

Signed-off-by: Jayati Shrivastava <gaurijove@gmail.com>

### CODE REVIEW

File: saskatoon/harvest/signals.py
Comment (2ynn) [reply saskatoon/harvest/signals.py:49]: yes ideally it should never happen that an `Actor` is neither a `Person` nor an `Organization` but we do have 3 "Unknown Actors" in the database right now. So yes I agree it'd be better to catch if `instance.owner.get_organization()` returns None.

All that being said @tristanlatr I think I have been very confused with this and I think I have implemented many things where I assume a property owner is a `Person` and I didn't realize an `Organization` could also have a property (we have 16 in the database right now). So I think I should open a separate issue for this and we should adapt the "register owner" workflow to the case where it is an organization.

Comment (tristanlatr) [reply saskatoon/harvest/signals.py:49]: Sounds good

File: saskatoon/harvest/signals.py
Comment (tristanlatr) [inline saskatoon/harvest/signals.py:56]: There are three unknowns properties that will make this block execute If I understand correctly.
I’d suggest we simply return None instead of crashing.

**Review by tristanlatr — APPROVED** [review]
Hi @victorphoenix3, thanks a lot and sorry for the late reply.

I think it looks good and can be merged. Did you had a change to configure saskatoon email alerts on your local computer ?

I'll let @2ynn merge this PR, when you'r ready.

Talk to you later

**Comment by victorphoenix3** [comment]
I will update this PR as per https://github.com/LesFruitsDefendus/saskatoon-ng/pull/213

File: saskatoon/harvest/models.py
Comment (tristanlatr) [inline saskatoon/harvest/models.py:777]: Just a little question, why using `pre_save` and not `post_save` here? 
If there is an error while calling `save()`, I suppose the email would have been triggered anyway.

Tell me what you think,

Comment (2ynn) [reply saskatoon/harvest/models.py:777]: I think `pre_save` is used to detect the change in status (`original_instance.pending`). After saving the object you'd loose track of what the former value of `pending` used to be. An alternative to avoid relying on signals would have been to manually store `self.pending` in the model's `__init__` and compare it to the new value in the `save()` method but i find the `pre_save` pretty elegant.

Comment (tristanlatr) [reply saskatoon/harvest/models.py:777]: ok very well, this was a misunderstanding.

**Comment by victorphoenix3** [comment]
@2ynn shouldn't this PR be merged into the branch `Fix-188-Not_all_owners_are_persons` since it depends on methods that were recently added to this branch and which are not yet integrated into the `develop` branch?

**Comment by 2ynn** [comment]
@victorphoenix3 Ugh my bad, you clearly indicated it in first comment. I just rebased `Fix-188-Not_all_owners_are_persons` onto develop to at least include your changes

**Pull request merged**
